### PR TITLE
Add _flushLimit and _flushInterval properties check to Logger. Use them if they exist. 

### PR DESCRIPTION
**Pull request by vilyapilya**
https://github.com/logdna/nodejs/issues/50

Addresses this enhancement request. 
A user can add `flushLimit` and `flushInterval` property to the option which is passed to the Logger constructor.  The Logger will use these properties if they are passed.

### CODE REVIEW

**Comment by smusali** [comment]
@vilyapilya, can you set the title more descriptive + add the description by including and mentioning the issue (with hyperlink or `#`)? Thanks in advance!

**Comment by smusali** [comment]
@vilyapilya, it seems like there are lint issues - since this project has its own `.eslintrc`, do `eslint --fix ...` first, then push the changes. Thanks in advance!

File: lib/logger.js [hunk 1]
```diff
@@ -99,6 +99,16 @@ function Logger(key, options) {
...
+        this._flushLimit = configs.FLUSH_BYTE_LIMIT;
+    }
+
+    if (options.flushInterval && Number.isInteger(options.flushInterval)) {
+        this._flushInterval = options.flushInterval;
+    }
+
     this._max_length = options.max_length || true;
     this._index_meta = options.index_meta || false;
     this._flushLimit = configs.FLUSH_BYTE_LIMIT;
```
Comment (smusali) [inline lib/logger.js]: If you set this up differently above, make sure you remove this line.

File: lib/logger.js [hunk 2]
```diff
@@ -99,6 +99,16 @@ function Logger(key, options) {
...
     }
 
+    if (options.flushLimit && Number.isInteger(options.flushLimit)) {
+        this._flushLimit = options.flushLimit;
+    } else {
+        this._flushLimit = configs.FLUSH_BYTE_LIMIT;
+    }
+
+    if (options.flushInterval && Number.isInteger(options.flushInterval)) {
+        this._flushInterval = options.flushInterval;
```
Comment (smusali) [inline lib/logger.js:109]: Set `this._flushInterval` in the same way you set `this._flushLimit`

File: lib/logger.js [hunk 3]
```diff
@@ -215,7 +225,7 @@ Logger.prototype._bufferLog = function(message) {
             if (err) {
                 debug('Received an error while flushing...' + err);
             }
-        }), configs.FLUSH_INTERVAL);
+        }), this._flushInterval || configs.FLUSH_INTERVAL);
```
Comment (smusali) [inline lib/logger.js]: Once you set `this._flushInterval` above by adding `else` condition, you won't need to use `||` here.

**Review by smusali — CHANGES_REQUESTED** [review]
Few changes to make

File: lib/logger.js (hunk 2 above)
Comment (vilyapilya) [reply lib/logger.js:109]: I can do that, but the logic for this property is different.

**Review by smusali — APPROVED** [review]

**Pull request merged**